CompetitionTemplate/src/competition/subsystems/vision/JetsonCommPacket.java

```diff
@@ -0,0 +1,190 @@
+                if(this.expectedPayloadLength > 0) {
+                    currentParserState = PacketParserState.WAITING_FOR_PAYLOAD_DATA;
+                }
+                else if (this.expectedPayloadLength == 0) {
+                    currentParserState = PacketParserState.PARSE_COMPLETE;
+                    
+                    return false;
+                }
+                else {
+                    currentParserState = PacketParserState.MALFORMED_PACKET_ABORT;
+                    // If we get junky data, give up
+                    return false;
+                }
+                break;
+
+            case WAITING_FOR_PAYLOAD_DATA:
+                this.packetPayloadData.add((Integer) newValue);
+                
+                if (packetPayloadData.size() >= expectedPayloadLength) {
+                    currentParserState = PacketParserState.PARSE_COMPLETE;
+                    return false;
+                }
+                break;
+            case PARSE_COMPLETE:
+                // We shouldn't get here; consumer should stop after all data is received
+                return false;
+            case MALFORMED_PACKET_ABORT:
+                return false;
+            default:
+                // Do nothing
+        }
+
+        return true;
+    }
+    
+    public boolean addNewValues(int[] newValues) {
+        if(newValues == null || newValues.length <= 0) {
+            log.error("No values given to add!");
+            return !isParseComplete();
+        }
+        
+        boolean shouldContinue = true;
+        for(int i = 0;
+                i < newValues.length && (shouldContinue = addNewValue(newValues[i]));
+                i++) {
+            continue;
+        }
+        
+        return shouldContinue;
+    }
+    
+    public boolean isParseComplete() {
+        return this.currentParserState == PacketParserState.PARSE_COMPLETE
+                || this.currentParserState == PacketParserState.MALFORMED_PACKET_ABORT;
+    }
+
+    @Override
+    public String toString() {
+        String result = "JetsonCommPacket [" + this.currentParserState + ", " + this.payloadType + "]";
+
+        result += " { ";
+        for (int i : this.packetPayloadData) {
+            result += i + " ";
+        }
+        result += "}";
+
+        return result;
+    }
+
+    public PacketPayloadType getPayloadType() {
+        return this.currentParserState == PacketParserState.MALFORMED_PACKET_ABORT ? PacketPayloadType.UNKNOWN
+                : this.payloadType;
+    }
+
+    public PacketParserState getCurrentParserState() {
+        return this.currentParserState;
+    }
+
+    public int[] getPayloadData() {
+        return this.packetPayloadData.stream().mapToInt(i->i).toArray();
+    }
+    
+    public double getReciptTimestamp() {
+        return this.reciptTimestamp;
+    }
+    
+    public void setReciptTimestamp(double value) {
+        this.reciptTimestamp = value;
+    }
+
+    public enum PacketParserState {
```

Comment:
Would maybe move these enums to the top, but I could be wrong about conventions there